docs/resources/resource_template.md

```diff
@@ -0,0 +1,4321 @@
+- `spec` (Block List, Max: 1) Specification of the resource template (see [below for nested schema](#nestedblock--spec))
+
+### Optional
+
+- `timeouts` (Block, Optional) (see [below for nested schema](#nestedblock--timeouts))
+
+<a id="nestedblock--metadata"></a>
+### Nested Schema for `metadata`
+
+***Required***
+
+- `name` (String) name of the resource
+- `project` (String) Project of the resource
+
+***Optional***
+
+- `description` (String) description of the resource
+
+
+<a id="nestedblock--spec"></a>
+### Nested Schema for `spec`
+
+***Required***
+
+- `provider` (String) Specify the resource template provider
+- `repository_options` (Block List, Max: 1) Repository options to be provided (see [below for nested schema](#nestedblock--spec--repository_options))
+- `version` (String) Version of the resource template
+
+***Optional***
+
+- `agents` (Block List) Reference to agents to process resource template (see [below for nested schema](#nestedblock--spec--agents))
+- `contexts` (Block List) Reference to config context data associated with resource templates (see [below for nested schema](#nestedblock--spec--contexts))
+- `hooks` (Block List, Max: 1) Lifecycle hooks for resource template (see [below for nested schema](#nestedblock--spec--hooks))
+- `provider_options` (Block List, Max: 1) Provider specific options to be provided (see [below for nested schema](#nestedblock--spec--provider_options))
+- `sharing` (Block List, Max: 1) Sharing resource template across projects (see [below for nested schema](#nestedblock--spec--sharing))
+- `variables` (Block List) Variables associated with resource template (see [below for nested schema](#nestedblock--spec--variables))
+- `version_state` (String) Represents the current state of template version
+
+<a id="nestedblock--spec--agents"></a>
+### Nested Schema for `spec.agents`
+
+***Required***
+
+- `name` (String) name of the resource
+
+
+<a id="nestedblock--spec--contexts"></a>
+### Nested Schema for `spec.contexts`
+
+***Required***
+
+- `name` (String) name of the resource
+
+
+<a id="nestedblock--spec--hooks"></a>
+### Nested Schema for `spec.hooks`
+
+***Optional***
+
+- `on_completion` (Block List) Configure the on completion lifecycle hook (see [below for nested schema](#nestedblock--spec--hooks--on_completion))
+- `on_failure` (Block List) Configure the on failure lifecycle hook (see [below for nested schema](#nestedblock--spec--hooks--on_failure))
+- `on_init` (Block List) Configure the on initialize lifecycle hook (see [below for nested schema](#nestedblock--spec--hooks--on_init))
+- `on_success` (Block List) Configure the on success lifecycle hook (see [below for nested schema](#nestedblock--spec--hooks--on_success))
+- `provider` (Block List, Max: 1) Configure the provider hooks (see [below for nested schema](#nestedblock--spec--hooks--provider))
+
+<a id="nestedblock--spec--hooks--on_completion"></a>
+### Nested Schema for `spec.hooks.on_completion`
```

Comment:
same comments as environment template

Reply:
updated relevant comments applicable for environment template as well